cocoa/src/toga_cocoa/widgets/selection.py

```diff
@@ -33,19 +32,49 @@ def rehint(self):
             max(self.interface._MIN_WIDTH, content_size.width)
         )
 
-    def remove_all_items(self):
+    def change_source(self, source):
         self.native.removeAllItems()
+        for item in source:
+            self.native.addItemWithTitle(self.interface._title_for_item(item))
+        self.interface.on_change(None)
```

Comment:
This method can be moved to the interface layer and implemented as follows:
* Disable on_change
* impl.clear
* impl.insert each item
* Enable on_change
* Fire on_change